lib/circuits_sim/device/sht4x.ex

```diff
@@ -0,0 +1,88 @@
+defmodule CircuitsSim.Device.SHT4X do
+  @moduledoc """
+  Sensirion SHT4x sensors
+
+  Typically found at 0x44
+  See the [datasheet](https://cdn-learn.adafruit.com/assets/assets/000/099/223/original/Sensirion_Humidity_Sensors_SHT4x_Datasheet.pdf) for details.
+  Many features aren't implemented.
+  """
+  alias CircuitsSim.I2C.I2CDevice
+  alias CircuitsSim.I2C.I2CServer
+  alias CircuitsSim.Tools
+
+  defstruct current: 0
+  @type t() :: %__MODULE__{current: non_neg_integer()}
+
+  @spec child_spec(keyword()) :: Supervisor.child_spec()
+  def child_spec(args) do
+    device = __MODULE__.new()
+    I2CServer.child_spec_helper(device, args)
+  end
+
+  @spec new() :: %__MODULE__{current: 0}
+  def new() do
+    %__MODULE__{}
+  end
+
+  ## protocol implementation
+
+  defimpl I2CDevice do
+    @cmd_serial_number 0x89
+    @cmd_measure_high_repeatability 0xFD
+    @cmd_measure_medium_repeatability 0xF6
+    @cmd_measure_low_repeatability 0xE0
+
+    @impl I2CDevice
+    def read(state, _count) when state.current == @cmd_serial_number do
+      {<<15, 186, 124, 249, 143, 14>>, %{state | current: 0}}
+    end
+
+    def read(state, _count) when state.current == @cmd_measure_high_repeatability do
+      {<<105, 234, 13, 109, 6, 50>>, %{state | current: 0}}
+    end
+
+    def read(state, _count) when state.current == @cmd_measure_medium_repeatability do
+      {<<105, 234, 13, 108, 239, 143>>, %{state | current: 0}}
+    end
+
+    def read(state, _count) when state.current == @cmd_measure_low_repeatability do
+      {<<105, 219, 249, 108, 200, 158>>, %{state | current: 0}}
+    end
+
+    def read(state, _count) do
+      {<<0, 0, 0, 0, 0, 0>>, %{state | current: 0}}
+    end
+
+    @impl I2CDevice
+    def write(state, <<@cmd_serial_number>>) do
+      %{state | current: @cmd_serial_number}
+    end
+
+    def write(state, <<@cmd_measure_high_repeatability>>) do
+      %{state | current: @cmd_measure_high_repeatability}
+    end
+
+    def write(state, <<@cmd_measure_medium_repeatability>>) do
+      %{state | current: @cmd_measure_medium_repeatability}
+    end
+
+    def write(state, <<@cmd_measure_low_repeatability>>) do
+      %{state | current: @cmd_measure_low_repeatability}
+    end
+
+    def write(state, _), do: state
+
+    @impl I2CDevice
+    def write_read(state, _, _), do: state
```

Comment:
```suggestion
    def write_read(state, _, _), do: {<<>>, state}
```

I'm slightly sad that Dialyzer didn't catch this.